### PR TITLE
Split protocol/ledger related concerns.

### PR DESCRIPTION
**Pull request by nc6**
This is not ready for merging, since it needs references added, but should serve
to start guiding the implementation at the Haskell level.

### CODE REVIEW

File: specs/chain/latex/blockchain-spec.tex [hunk 1]
```diff
@@ -626,31 +612,158 @@ \section{Counting signed blocks}
...
+\newcommand{\bhhash}[1]{\fun{\bhhashname}\ #1}
+\newcommand{\bhprevhashname}{bhPrevHash}
+\newcommand{\bhprevhash}[1]{\fun{\bhprevhashname}\ #1}
+\newcommand{\bhtosignname}{bhToSign}
+\newcommand{\bhtosign}[1]{\fun{\bhtosignname}\ #1}
+
+During PBFT processing of the block header, we do the following:
+\begin{enumerate}
+ \item We check that the current block is being issued for a slot later than
+    the last block issued.
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex]: maybe "later than the slot of the last block issued"? Or "later than the slot for which the last block was issued"?

File: specs/chain/latex/blockchain-spec.tex [hunk 2]
```diff
@@ -626,31 +612,158 @@ \section{Counting signed blocks}
...
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Abstract functions}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \fun{\bhprevhashname} & \Bhead \totalf \Hash & \text{Previous header hash} \\
+      \fun{\bhhashname} & \Bhead \totalf \Hash & \text{Header hash} \\
+      \fun{\bsigname} & \Bhead \totalf \Sig & \text{Block signature} \\
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex]: Maybe replace `bSig` by `bhSig` for consistency?

File: specs/chain/latex/blockchain-spec.tex [hunk 3]
```diff
@@ -626,31 +612,158 @@ \section{Counting signed blocks}
...
+      \fun{\bissuername} & \Bhead \totalf \VKey & \text{Block issuer} \\
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex]: Replace `bIssuer` with `bhIssuer`?

File: specs/chain/latex/blockchain-spec.tex [hunk 4]
```diff
@@ -626,31 +612,158 @@ \section{Counting signed blocks}
...
+      \fun{\bhslotname} & \Bhead \totalf \Slot & \text{Slot for which this block is issued}
+    \end{array}
+  \end{equation*}
+  \caption{Progressive BFT types and functions}
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex]: Permissive?

File: specs/chain/latex/blockchain-spec.tex [hunk 5]
```diff
@@ -626,31 +612,158 @@ \section{Counting signed blocks}
...
+    \trans{pbft}{\var{bh}}
+    \left(
+      {\begin{array}{c}
+         \bhhash{bh} \\
+         \var{sgs}'
+       \end{array}}
+   \right)
+ }
+\end{equation*}
+\caption{Progressive BFT rules}
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex]: I just realized that progressive is consistently used in the rest of the figures, and now I'm now sure whether you actually meant "progressive", or you we're listening to some progressive rock while typing this :P 

On a related note I'm thinking that "permissive-rock" does not exist, but it should.

**Review by dnadales — APPROVED** [review]
Getting better and better every day :) 

One question that I have is whether `EBB`'s are Genesis specific or are they also present in Praos. If so we might have to include them in the `PBFT` rule only.

**Comment by mdimjasevic** [comment]
What exactly should I take from this PR to work on issue #304?

For example, I see the field `bHeaderSize` is used in the `BHEAD` rule in the LaTeX version, but it is not present in Haskell in the `Cardano.Spec.Chain.STS.Block.BlockHeader` type. Should it be?

**Comment by mdimjasevic** [comment]
> What exactly should I take from this PR to work on issue #304?


@nc6, Is it the split into these two rules: `PBFT` and `BHEAD`?

File: specs/chain/latex/blockchain-spec.tex (hunk 5 above)
Comment (nc6) [reply specs/chain/latex/blockchain-spec.tex]: I wasn't, but am tempted to put on some Jethro Tull now...

Comment (dnadales) [reply specs/chain/latex/blockchain-spec.tex]: I keep on learning from you Nick :)

**Pull request merged**